41474 Update and restructure project files

mohmed98 opened the pull request:
# The Issue

## Issue Details

### Issue Reference
Fixes: https://github.com/elegantthemes/Divi/issues/41474

### Technical Description

- The old file structure was outdated. 
- `ObjectRenderer` was giving undefiled.


# The Pull Request

## Resolution Details

### PR Solution Explanation
- Add same plugin structure as [Create Divi 5 Module in 10 Minutes](https://devalpha.elegantthemes.com/docs/tutorials/module/beginner/create-simple-quick-module/)
- Edit webpack configuration to output two files because `registerBuilderBarButton` needs `'in_footer' => true` when enqueuing script in the `register_package_build` function
- Removed the usage of `window.divi.objectRenderer` as was giving error and couldn't figure out the problem I have implemented an alternative and will create new task to inspect it.



### Screencast Verification


https://github.com/user-attachments/assets/378bbcb4-6e54-4cd3-ae68-a38632c1c314




### QA Strategy for Regression Testing
- Make sure everything works as expected in the plugin.
- Button and Modal are opened.

### Changelog Copy
Updated plugin structure and needed code to run with the D5.

robicse11127 reviewed (approved):
LGTM!

ayubadiputra commented on src/add-bar-builder-buttons.js:
```diff
@@ -0,0 +1,18 @@
+import {
+  dispatch,
+} from '@divi/data';
+  window.divi.appUi.registerBuilderBarButton({
+    active: false,
+    iconSvg: { name: 'divi/move' },
+    label: 'My Second Button',
+    order: 20,
+    name: 'my-second-button',
+    onClick: () => {
+      dispatch('divi/modal-library').open({
+        name: 'divi/dev-clipboard',
+      });
+    },
+  });
```
```suggestion

window.divi.appUi.registerBuilderBarButton({
  active: false,
  iconSvg: { name: 'divi/move' },
  label: 'My Second Button',
  order: 20,
  name: 'my-second-button',
  onClick: () => {
    dispatch('divi/modal-library').open({
      name: 'divi/dev-clipboard',
    });
  },
});z
```

ayubadiputra replied:
**answer**: They are fixed here [feat: Add SCSS styling for modal component and integrate with build process](https://github.com/elegantthemes/d5-extension-example-modal-dev-clipboard/pull/5/files#top). You can ignore.

ayubadiputra commented on src/index.jsx:
```diff
@@ -0,0 +1,3 @@
+
+
```
**todo**: Remove those unused new lines.

ayubadiputra replied:
**answer**: They are fixed here [feat: Add SCSS styling for modal component and integrate with build process](https://github.com/elegantthemes/d5-extension-example-modal-dev-clipboard/pull/5/files#top). You can ignore.

ayubadiputra commented on src/modal/container.jsx:
```diff
@@ -8,6 +8,10 @@ const {
-}))(DevClipboard);
+export const DevClipboardContainer = withSelect((selectStore) => {
+  const clipboardItems = selectStore('divi/clipboard').getItems();
+  console.log('Clipboard Items:', DevClipboard);
```
**todo**: I think this should be removed.

ayubadiputra commented on src/modal/registerDevClipboard.js:
```diff
@@ -0,0 +1,12 @@
+import { DevClipboardContainer } from "./container";
+window.vendor.wp.hooks.addFilter('divi.modalLibrary.modalMapping', 'divi', modals => {
```
**todo**: Let's add new line here.

ayubadiputra reviewed (changes requested):
@mohmed98 Some small feedback. I understand that you have series of PR. Just let me know if somehow the feedback is addressed on somewhere else.

And also, I don't see the content on the modal. Probably I missed something.


https://github.com/user-attachments/assets/9f0b5b7b-a252-4d96-96d5-84425ca3eca8

ayubadiputra commented on src/add-bar-builder-buttons.js:
```diff
@@ -0,0 +1,18 @@
+  window.divi.appUi.registerBuilderBarButton({
+    active: false,
+    iconSvg: { name: 'divi/move' },
+    label: 'My Second Button',
```
**answer**: They are fixed here [feat: Add SCSS styling for modal component and integrate with build process](https://github.com/elegantthemes/d5-extension-example-modal-dev-clipboard/pull/5/files#top). You can ignore.

ayubadiputra commented on src/add-bar-builder-buttons.js:
```diff
@@ -0,0 +1,18 @@
+    iconSvg: { name: 'divi/move' },
+    label: 'My Second Button',
+    order: 20,
+    name: 'my-second-button',
```
**answer**: They are fixed here [feat: Add SCSS styling for modal component and integrate with build process](https://github.com/elegantthemes/d5-extension-example-modal-dev-clipboard/pull/5/files#top). You can ignore.

mohmed98 commented:
> @mohmed98 Some small feedback. I understand that you have series of PR. Just let me know if somehow the feedback is addressed on somewhere else.
> 
> And also, I don't see the content on the modal. Probably I missed something.
> 
>  No.content.on.modal.mp4

It's still working on my side, you copied something, right?
![image](https://github.com/user-attachments/assets/40b0ed63-c0ee-4ab6-9e15-ac4e61bf89cd)

mohmed98 commented:
@ayubadiputra I have merged:
- `41497-add-custom-icon`
- `41495-support-styling`
- `41496-update-oudtaed-packges`
In this one to make it easier to review, and addressed all feedback

mohmed98 commented:
> @mohmed98 Despite the small feedback which I think I can simply address. I still have problem where the modal content is always empty.
> 
> ![Modal Empty](https://private-user-images.githubusercontent.com/9313128/410838965-979d4309-c883-48bd-86cf-5cc0f4d2545e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg5MjE3MjIsIm5iZiI6MTczODkyMTQyMiwicGF0aCI6Ii85MzEzMTI4LzQxMDgzODk2NS05NzlkNDMwOS1jODgzLTQ4YmQtODZjZi01Y2MwZjRkMjU0NWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMDdUMDk0MzQyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzBhNWZjNTNlZjBhZjFlOGNmY2Y4NzliMDgzZTY0ZTUyZDg3OGY4NjRiYzNiZWIzOWE1MDQyNGZhYTRmNmQ4NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.pawcAwMqO0TlbmqUXpsrf0UYnJLL2N063V0T1AEL3LU)

This has been resolved in a call, we needed to copy something

ayubadiputra reviewed (approved):
LGTM 👍
